src/core/http1/client.rs

```diff
@@ -0,0 +1,604 @@
+/*
+ * Copyright (C) 2020-2023 Fanout, Inc.
+ * Copyright (C) 2023-2024 Fastly, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::buffer::{Buffer, VecRingBuffer, VECTORED_MAX};
+use crate::core::http1::error::Error;
+use crate::core::http1::util::*;
+use crate::future::{
+    select_2, AsyncRead, AsyncWrite, AsyncWriteExt, ReadHalf, Select2, StdWriteWrapper, WriteHalf,
+};
+use crate::http1;
+use crate::pin;
+use std::cell::RefCell;
+use std::io::{self, Write};
+use std::mem;
+use std::pin::Pin;
+use std::str;
+
+pub struct Request<'a, R: AsyncRead, W: AsyncWrite> {
+    r: ReadHalf<'a, R>,
+    w: WriteHalf<'a, W>,
+    buf1: &'a mut VecRingBuffer,
+    buf2: &'a mut VecRingBuffer,
```

Comment:
I would recommend naming these to `inputBuf` & `outputBuf` for clarity

Reply:
since I was only meaning to move code, let's leave this to a separate PR?